ex21.py

```diff
@@ -51,12 +54,17 @@ def divide(a, b):
 
 # assigning the variable "what" to the return of....
 # fuck I wrote all this backwards ><
-# calling the "divide()" function using 2 and 3 as its arguments, a and b. Then 
+# calling the "divide()" function using 2 and 3 as its arguments, a and b. Then

```

Comment:
This whole comment block is basically worthless as documentation, though it may have been a good exercise for thinking hard about how the nested functions are called.